src/describe/resource.ts

```diff
@@ -15,7 +15,7 @@ export type ResourceType<IdentityType, ResourceDataType> = {
     identity: IdentityType,
   ) => Promise<ResourceDataType>;
   UNSTABLE__clearImmediate: boolean;
-  use: (identity: IdentityType) => [IdentityType | undefined, any];
+  use: (identity: IdentityType) => [ResourceDataType | undefined, any];
```

Comment:
```suggestion
  use: (identity: IdentityType) => [ResourceDataType[] | null[] | undefined, number | undefined, any];
```

Reply:
Hmm, just want to double check, but the change you are suggesting seems like it would be good for the PaginatedResource and not the normal one, is that right? 🤔

Reply:
Added a commit for what think you want versus what was suggested: https://github.com/AlphaFlow/resource/pull/13/commits/74cd7f5a90f50b20d46bf1c46f125a813d2ca842

Also, I noticed there is a bunch of usage of `.use()` w/o any args. in the `admin-client` so I assume we also want to allow for that (unless we want to users to intentionally call `.use(undefined)` which would also be fine. Commit for that: https://github.com/AlphaFlow/resource/pull/13/commits/7c19487a9338485319ed375d93fbeb5ef8f1703d

Reply:
> Hmm, just want to double check, but the change you are suggesting seems like it would be good for the PaginatedResource and not the normal one, is that right? 🤔

You're right... I'm not sure why I recommended this, I may have just not realized I was looking at the normal resource file.